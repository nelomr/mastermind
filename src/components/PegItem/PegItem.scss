.peg-item {
  width: 10px;
  height: 10px;
  margin: 5px 5px;
  border: 1px solid transparent;
  border-radius: 50%;

  &.is-white {
    border-color: $black-color;
  }
}