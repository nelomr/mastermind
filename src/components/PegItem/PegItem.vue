<template>
  <div 
    :class="{'is-white': newColor === pegsColorOption}"
    :style="`background-color: ${ newColor };`"
    class="peg-item"
  >

  </div>
</template>
<script>
import { mapState } from 'vuex';

export default {
  name: 'PegsItem',
  props: {
    color: {
      type: String,
      default: 'lightgray'
    }
  },
  data() {
    return {
      currentColor: this.color
    }
  },
  computed: {
    ...mapState({
      pegsColorOption: state => state.pegsColorOption.partial
    }),
    newColor() {
      return this.color;
    }
  }
}
</script>
<style lang="scss">
  @import "./PegItem.scss";
</style>