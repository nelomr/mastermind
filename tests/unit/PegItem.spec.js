import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import PegItem from '@/components/PegItem/PegItem.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('PegItem.vue', () => {
  let color = 'lightgray';
  let pegsColorOption = 'graylight';
  let props;
  let store;
  let state;

  beforeEach(() => {
    props = {
      color
    };
    state = {
      pegsColorOption
    };
    store = new Vuex.Store({
      state
    });
  });
  const build = () => {
    const wrapper = shallowMount(PegItem, {
      props,
      mocks: {$store: store}
    });

    return {
        wrapper
    };
  };

  it('renders the component', () => {
    const {wrapper} = build();

    expect(wrapper.html()).toMatchSnapshot();
  });
})
