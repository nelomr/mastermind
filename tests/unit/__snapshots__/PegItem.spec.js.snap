// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PegItem.vue renders the component 1`] = `<div class="peg-item is-white"></div>`;
